src/lambda/wizards/samInitWizard.ts

```diff
@@ -169,11 +181,18 @@ export class CreateNewSamAppWizard extends MultiStepWizard<SamCliInitArgs> {
     private runtime?: lambdaRuntime.SamLambdaRuntime
     private location?: vscode.Uri
     private name?: string
+    private readonly context: CreateNewSamAppWizardContext
 
     public constructor(
-        private readonly context: CreateNewSamAppWizardContext = new DefaultCreateNewSamAppWizardContext()
+        private readonly vscodeContext: Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'>,
```

Comment:
This is still part of the context for `CreateNewSamAppWizard`, so it shouldn't be a separate argument.

### Option 1: Make the vscode context a property of the wizard context

```typescript
export interface CreateNewSamAppWizardContext {
    ...
    vscodeContext: Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'>
}
```

### Option 2: Merge the vscode context into the wizard context

```typescript
interface CreateNewSamAppWizardContext { /* unchanged */ }
type MergedContext = CreateNewSamAppWizardContext & Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'>
export { MergedContext as CreateNewSamAppWizardContext }

```

### Option 3: Make `CreateNewSamAppWizardContext` derive from the extension context.

```typescript
export interface CreateNewSamAppWizardContext implements Pick<vscode.ExtensionContext, 'asAbsolutePath' | 'globalState'> { /* unchanged */ }
```